src/features/groups/Members/index.tsx

```diff
@@ -1,61 +1,100 @@
-import { memo, useCallback } from 'react';
+import { memo, useCallback, useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 
 import Button from '../../../components/Button';
 import { Column, Row } from '../../../components/Containers';
 import { TextRegular } from '../../../components/Text';
-import AddAvatarSVG from '../../../assets/images/add-avatar.svg';
-import { Avatar, ContentWrapper } from './index.styled';
-import { useGetTransactionGroupByIdQuery } from '../api';
-import { getDownloadFileUrl } from '../../../helpers/urlHelper';
+import { ContentWrapper, RemoveBtn, Wrapper } from './index.styled';
+import { useAddParticipantMutation, useGetTransactionGroupByIdQuery, useRemoveParticipantMutation } from '../api';
+import { useModal } from '../../../providers/ModalProvider';
+import AddMemberModal from './AddMemberModal';
+import { IUser } from '../../../types/user-types';
+import Loader from '../../../components/Loader';
+import MemberRow from './MemberRow';
 
 export interface IGroupMembersProps {
-  groupId: string
+  groupId: string,
+  onMemberAdded: (member: IUser) => void;
+  onMemberRemoved: (member: IUser) => void;
 }
 
-const GroupMembers = ({ groupId }: IGroupMembersProps) => {
+const GroupMembers = ({ groupId, onMemberAdded, onMemberRemoved }: IGroupMembersProps) => {
   const { t } = useTranslation('groups');
 
   const { data } = useGetTransactionGroupByIdQuery(groupId);
+  const [addMember, { isLoading: isMemberAdding }] = useAddParticipantMutation();
+  const [removeMember, { isLoading: isMemberRemoving }] = useRemoveParticipantMutation();
+  const [addMemberModalId, setAddMemberModalId] = useState<number | null>(null);
+
+  const [selected, setSelected] = useState<IUser | null>(null);
+
+  const handleMemberToAddSelected = useCallback((member: IUser) => {
+    addMember({ groupId: groupId, userId: member.id }).then(() => onMemberAdded(member));
+  }, [addMember, groupId, onMemberAdded]);
+
+  const handleMemberSelected = useCallback((member: IUser) => {
+    setSelected(member);
+  }, [setSelected]);
+
+  const modalContext = useModal();
 
   const onAddMember = useCallback(() => {
+    const modalId = modalContext.showModal(<AddMemberModal onMemberSelected={handleMemberToAddSelected} />);
+    setAddMemberModalId(modalId);
+  }, [modalContext, handleMemberToAddSelected, setAddMemberModalId]);
+
+  const onRemoveMember = useCallback(() => {
+    removeMember({ groupId, userId: selected!.id }).then(() => {
+      onMemberRemoved(selected!);
+      setSelected(null);
+    });
+  }, [groupId, selected, removeMember, onMemberRemoved]);
 
-  }, []);
+  useEffect(() => {
+    return () => {
+      if (addMemberModalId) {
+        modalContext.closeModal(addMemberModalId);
+        setAddMemberModalId(null);
+      }
+    };
+  }, [addMemberModalId, modalContext]);
```

Comment:
Try to test with addMemberModalId as ref

Reply:
Ref cause modal to be closed after it opened